### PR TITLE
Add tun.unavailable_reason()

### PR DESCRIPTION
**Pull request by errorxyz**

### CODE REVIEW

File: mitmproxy-rs/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ tokio = { version = "1.41", features = ["macros", "net", "rt-multi-thread", "syn
 boringtun = "0.6"
 tar = "0.4.43"
 console-subscriber = { version = "0.4.1", optional = true }
-
+nix = "0.17.0"
```
Comment (decathorpe) [inline mitmproxy-rs/Cargo.toml:32]: Is there a reason why you added such an old version of `nix` here? The 0.17 release is from 2020, the current version is 0.29.

Comment (errorxyz) [reply mitmproxy-rs/Cargo.toml:32]: oops, thanks for spotting it out

File: mitmproxy-rs/src/server/tun.rs [hunk 2]
```diff
@@ -59,6 +62,24 @@ pub fn create_tun_interface(
...
-        "TUN proxy mode is only available on Linux",
+        unavailable_reason(),
     ))
 }
+
+/// Determines the reason for unavailability of TUN proxy mode
+///
+/// Returns
+/// - `String`: reason for unavailability
+/// - `None`: if available or reason unknown for unavailability
```
Comment (mhils) [inline mitmproxy-rs/src/server/tun.rs:73]: These docstrings ends up in Python, so we should use Python syntax (str and not String). Let's also shorten this a bit.


```suggestion
/// Returns a `str` describing why tun mode is unavailable, or `None` if TUN mode is available.
/// 
/// Reasons for unavailability may be an unsupported platform, or missing privileges.
```

**Review by mhils — COMMENTED** [review]
Nice, thanks! LGTM % review comments.

**Review by mhils — APPROVED** [review]

**Pull request merged**